packages/core/src/components/popover/popover.tsx

```diff
@@ -551,24 +560,27 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
 
     private updateTether() {
         if (this.state.isOpen && !this.props.inline && this.popoverElement != null) {
+            const { constraints, position, tetherOptions, useSmartPositioning } = this.props;
+
             // the .pt-popover-target span we wrap the children in won't always be as big as its children
             // so instead, we'll position tether based off of its first child.
             // NOTE: use findDOMNode(this) directly because this.targetElement may not exist yet
             const target = findDOMNode(this).childNodes[0];
 
             // constraints is deprecated but must still be supported through tetherOptions until v2.0
-            if (this.props.constraints != null) {
-                this.props.tetherOptions.constraints = this.props.constraints;
-            }
-
-            const tetherOptions = TetherUtils.createTetherOptions(
-                this.popoverElement, target, this.props.position,
-                this.props.useSmartPositioning, this.props.tetherOptions,
-            );
+            const options = (constraints == null && !useSmartPositioning)
+                ? tetherOptions
+                : {
+                    ...tetherOptions,
+                    constraints: useSmartPositioning ? [SMART_POSITIONING] : constraints,
```

Comment:
note that this is consistent with the prior behavior, where `useSmartPositioning=true` would override all other `constraints`.